src/pydisagg/ihme/splitter/cat_splitter.py

```diff
@@ -0,0 +1,632 @@
+from typing import Any, List
+
+import numpy as np
+import pandas as pd
+import multiprocessing
+from pandas import DataFrame
+from pydantic import BaseModel
+from typing import Literal
+from joblib import Parallel, delayed
+
+from pydisagg.disaggregate import split_datapoint
+from pydisagg.ihme.schema import Schema
+from pydisagg.models import RateMultiplicativeModel, LogOddsModel
+from pydisagg.ihme.validator import (
+    validate_columns,
+    validate_index,
+    validate_noindexdiff,
+    validate_nonan,
+    validate_positive,
+)
+
+
+class CatDataConfig(Schema):
+    """
+    Configuration for the data DataFrame.
+
+    Parameters
+    ----------
+    index : List[str]
+        List of column names to be used as index in the data DataFrame.
+    target : str
+        Column name representing the target variable to split.
+    val : str
+        Column name for the observed value.
+    val_sd : str
+        Column name for the standard deviation of the observed value.
+    """
+
+    index: List[str]
+    target: str
+    val: str
+    val_sd: str
+
+    @property
+    def columns(self) -> List[str]:
+        """
+        List of all required columns in the data DataFrame.
+
+        Returns
+        -------
+        List[str]
+            List of column names.
+        """
+        return list(set(self.index + [self.target, self.val, self.val_sd]))
+
+    @property
+    def val_fields(self) -> List[str]:
+        """
+        List of value fields (attributes).
+
+        Returns
+        -------
+        List[str]
+            List containing 'val' and 'val_sd'.
+        """
+        return ["val", "val_sd"]  # Attribute names
+
+
+class CatPatternConfig(Schema):
+    """
+    Configuration for the pattern DataFrame.
+
+    Parameters
+    ----------
+    index : List[str]
+        List of column names to be used as index in the pattern DataFrame.
+    target : str
+        Column name representing the target variable to split.
+    draws : List[str], optional
+        List of draw column names, by default [].
+    val : str, optional
+        Column name for the mean value in the pattern DataFrame, by default 'mean'.
+    val_sd : str, optional
+        Column name for the standard deviation in the pattern DataFrame, by default 'std_err'.
+    prefix : str, optional
+        Prefix to apply to column names when merging, by default 'cat_pat_'.
+    """
+
+    index: List[str]
+    target: str
```

Comment:
And I think we should stick with the strategy we had with age splitter, have `by` and `target` (or `cat`), and have a property `index` to group `by` and `target`.